### PR TITLE
Reorganize examples

### PR DESCRIPTION
**Pull request by henry2004y**
Move examples into docs with WGLMakie support. Handle #71 and #74.

With these changes, each example takes a separate page and generates < 21 MB html file. We won't have outdated demos anymore!

### CODE REVIEW

**Comment by TCLiuu** [comment]
There is still one issue here, which is the warning in the log.
```
┌ Warning: cover file doesn't exists
│   cover_path = "/home/runner/work/TestParticle.jl/TestParticle.jl/docs/examples/simulations/../../figures/proton_ganymede_mhdepic.png"
└ @ DemoCards ~/.julia/packages/DemoCards/ZgH8l/src/generate.jl:358
```
This should be a bug in `DemoCards`, and I have already submitted the relevant issue.
https://github.com/JuliaDocs/DemoCards.jl/issues/146

File: docs/examples/basics/demo_magneticbottle.jl [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
+# author: "[Hongyang Zhou](https://github.com/henry2004y)"
+# julia: 1.9.0
+# description: Tracing charged particle in a magnetic bottle
+# ---
+
+# This example shows how to trace non-relativistic and relativistic electrons in a
+# stationary magnetic field that corresponds to a magnetic bottle.
 # Reference: https://en.wikipedia.org/wiki/Magnetic_mirror#Magnetic_bottles
-#
-# Hongyang Zhou, hyzhou@umich.edu
```
Comment (TCLiuu) [inline docs/examples/basics/demo_magneticbottle.jl:13]: Why aren't images exported here?

**Comment by TCLiuu** [comment]
Additionally, there is another issue where all the examples have been moved from the 'examples' folder to 'docs', and the dependency has been changed to `WGLMakie`. The slider on the web version cannot interact properly.

My question is, whether it is necessary to maintain a version of `GLMakie` for some examples. In my teaching, I directly run the examples and introduce them through interaction.

**Comment by TCLiuu** [comment]
There is one more strange thing, when running this command on my computer:
```
julia --project make.jl
```
There is a strange error in my log, which did not appear in the GitHub action.
```
Warning: failed to run `@example` block in src/examples/basics/demo_magneticmirror.md:14-109
│ ```@example demo_magneticmirror
│ using JSServe: Page # hide
│ Page(exportable=true, offline=true) # hide
│
│ using TestParticle
│ using TestParticle: getB_mirror
│ using OrdinaryDiffEq
│ using StaticArrays
│ using LinearAlgebra
│ using TestParticleMakie
│ using Printf
│ using WGLMakie
│
│ ### Obtain field
│
│ # Magnetic mirror parameters in SI units
│ const I = 20. # current in the solenoid
│ const N = 45 # number of windings
│ const distance = 10. # distance between solenoids
│ const a = 4.0 # radius of each coil
│
│ function getB(xu)
│    SVector{3}(getB_mirror(xu[1], xu[2], xu[3], distance, a, I*N))
│ end
│...
|...
│ f
│ ```
│   value = UndefVarError: `getB_mirror` not defined
└ @ Documenter.Expanders ~/.julia/packages/Documenter/H5y27/src/Utilities/Utilities.jl:34
```
Can I trouble you to help test and see if the same error occurs? @henry2004y

File: docs/examples/basics/demo_magneticbottle.jl (hunk 1 above)
Comment (henry2004y) [reply docs/examples/basics/demo_magneticbottle.jl:13]: Ah, because I forgot to add the JSServe Page settings here. Will be fixed.

**Comment by henry2004y** [comment]
> Can I trouble you to help test and see if the same error occurs? @henry2004y

I have a similar issue yesterday. `getB_mirror` is a recently added method (that lives in master but not in a release patch), so you need to use the master branch of TestParticle to make it work. What I did is

```julia
pkg> add TestParticle#master
```

After we register a new patch this is no longer an issue.

**Comment by TCLiuu** [comment]
I found that there are some display issues with `Democards`. This is how it looks in the version I generated locally.
image

It is different from the version of the documentation of `Democards` with theme "grid". This should not be a problem with my network causing the CSS to not load correctly. May I ask if your local version has the same issue?

**Comment by henry2004y** [comment]
I have the same issue, which does not happen for my other package [Vlasiator demo](https://henry2004y.github.io/Vlasiator.jl/dev/examples/). Based on a quick scan the json files are identical...

**Pull request merged**